### ISSUE TITLE
Allow loading configs from internal JAR resources via an InputStream

### ISSUE THREAD

**Issue by WiIIiam278** [Exlll/ConfigLib#27]
**Is your feature request related to a problem? Please describe.**
I'd like to include several user-provided language files as resources within the JAR files I distribute and load these, based on the user's entry in their main config file, into a `Configurable`, which I can then save to the user's config directory, update, etc.

**Describe the solution you'd like**
Expose methods in the various implementations that accept an `InputStream` for loading from internally bundled JAR resources. Perhaps even a utility `update` method that accepts "defaults" to be read from an internal resource.

Obviously, the typical (and nice) way of defining defaults for configuration is in-code - but in the case of large multi-language locale files where the defaults vary based on the user's language, a resource-based solution seems more appropriate.

**Describe alternatives you've considered**
Manually copying the resource from the plugin to the user's config folder, circumventing the use of your library, or delving into SnakeYAML internals to do this :-(

**Additional context**
_None_

**Comment by Exlll** [comment]
Hey, thanks for sharing your idea. How would you like to use such an API? From my understanding, you would like me to add the following two methods (and their overloads) to this library:

```java
YamlConfigurationStore.load(InputStream)
YamlConfigurations.load(InputStream, Class<T>)
```

Did I get this right or do you require something else?

**Comment by Exlll** [comment]
Thanks for taking the time to test it. :slightly_smiling_face: I've just published version `4.3.0` which contains these changes.

If you'd like anything else, let me know!
